### PR TITLE
[ACR] `az acr token create`: Fix random order of repo valid actions and gateway valid actions in the help message

### PR DESCRIPTION
**Pull request by mabenedi**
**Related command**
az acr token create -h

**Description**
Currently, the order of allowed actions listed in the help text is random and changes every time the -h is invoked. This causes problem for the docs people who are diffing the help text at regular intervals.

**Testing Guide**
az acr token create -h

**History Notes**

[ACR] `az acr token create`: Fix random order of repo valid actions and gateway valid actions in the help message

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ACR

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 1]
```diff
@@ -430,8 +430,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('description', options_list=['--description'], help='Description for the scope map. Maximum 256 characters are allowed.', required=False)
         c.argument('scope_map_name', options_list=['--name', '-n'], help='The name of the scope map.', required=True)
 
-    repo_valid_actions = "Valid actions are {}".format({action.value for action in RepoScopeMapActions})
-    gateway_valid_actions = "Valid actions are {}".format({action.value for action in GatewayScopeMapActions})
+    # Action strings generated this way to ensure consistent ordering each time the help text is generated.
+    repo_valid_actions = "Valid actions are {" + ', '.join('\'{}\''.format(action.value) for action in RepoScopeMapActions) + "}."
+    gateway_valid_actions = "Valid actions are {" + ', '.join('\'{}\''.format(action.value) for action in GatewayScopeMapActions) + "}."
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acr/_params.py:435]: Just curiosity, may I ask why this way can solve the problem of random order?

Comment (mabenedi) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:435]: I really don't understand why this way gives a defined order and the other way doesn't. I would have thought that the original code would have a defined order too, but it doesn't

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 2]
```diff
@@ -430,8 +430,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('description', options_list=['--description'], help='Description for the scope map. Maximum 256 characters are allowed.', required=False)
         c.argument('scope_map_name', options_list=['--name', '-n'], help='The name of the scope map.', required=True)
 
-    repo_valid_actions = "Valid actions are {}".format({action.value for action in RepoScopeMapActions})
-    gateway_valid_actions = "Valid actions are {}".format({action.value for action in GatewayScopeMapActions})
+    # Action strings generated this way to ensure consistent ordering each time the help text is generated.
+    repo_valid_actions = "Valid actions are {" + ', '.join('\'{}\''.format(action.value) for action in RepoScopeMapActions) + "}."
```
[the remark this answers is not in the document]
Comment (mabenedi) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py]: Nice suggestion, I like your way better. Changing to that method

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 3]
```diff
@@ -430,8 +430,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('description', options_list=['--description'], help='Description for the scope map. Maximum 256 characters are allowed.', required=False)
         c.argument('scope_map_name', options_list=['--name', '-n'], help='The name of the scope map.', required=True)
 
-    repo_valid_actions = "Valid actions are {}".format({action.value for action in RepoScopeMapActions})
-    gateway_valid_actions = "Valid actions are {}".format({action.value for action in GatewayScopeMapActions})
+    # Action strings generated this way to ensure consistent ordering each time the help text is generated.
+    repo_valid_actions = "Valid actions are {}".format(sorted(list(action.value for action in RepoScopeMapActions)))
```
[the remark this answers is not in the document]
Comment (mabenedi) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py]: sure this is fine too, fixed

**Comment by mtrilbybassett** [comment]
> **Related command** az acr token create -h
> 
> **Description** Currently, the order of allowed actions listed in the help text is random and changes every time the -h is invoked. This causes problem for the docs people who are diffing the help text at regular intervals.
> 
> **Testing Guide** az acr token create -h
> 
> **History Notes**
> 
> [acr] `az acr token create -h`: Make help text consistent
> 
> This checklist is used to make sure that common guidelines for a pull request are followed.
> 
> * [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).
> * [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).
> * [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

Looking at other PRs, the format of the checkboxes might be the issue?

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 4]
```diff
@@ -430,8 +430,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('description', options_list=['--description'], help='Description for the scope map. Maximum 256 characters are allowed.', required=False)
         c.argument('scope_map_name', options_list=['--name', '-n'], help='The name of the scope map.', required=True)
 
-    repo_valid_actions = "Valid actions are {}".format({action.value for action in RepoScopeMapActions})
-    gateway_valid_actions = "Valid actions are {}".format({action.value for action in GatewayScopeMapActions})
+    # Action strings generated this way to ensure consistent ordering each time the help text is generated.
+    repo_valid_actions = "Valid actions are {}".format(sorted(action.value for action in RepoScopeMapActions))
```
[the remark this answers is not in the document]
Comment (mabenedi) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:434]: Sorry, what difference to the output do the braces make?

[the remark this answers is not in the document]
Comment (mabenedi) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:434]: I see. There won't be duplicate values in the enum

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@xmdanni Could you please help review this PR again?

**Pull request merged**